### ISSUE TITLE
Clippy misses integer arithmetic error for a specific type

### ISSUE THREAD

**Issue by Chralt98** [rust-lang/rust-clippy#8968]
### Summary

I expected an clippy error when using a specific type, which resolves to be an integer (`u128`).

### Reproducer

I tried this code:

```rust
pub type SpecificIntegerType = u128;
pub const NewValue: SpecificIntegerType = 42;

...

let test_arithmetic_integer_error = NewValue + NewValue;
```

When running with `-Dclippy::integer_arithmetic`

I expected to see this happen:

`error: integer arithmetic detected`

Instead, this happened:

No error.

### Version

```text
rustc 1.62.0-nightly
```

**Comment by Alexendoo** [comment]
It appears to trigger the lint for https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5ae8ae28263bb35ccc1987355557aa80

Is there some other context needed?

**Comment by Jarcho** [comment]
`integer_arithmetic` can't lint on generic types as other types could be substituted there.

**Comment by Chralt98** [comment]
Thank you for your answer. What a pitty!

**Comment by Jarcho** [comment]
There should be a lint for using any arithmetic operators soonish.

**Comment by c410-f3r** [comment]
With https://github.com/rust-lang/rust-clippy/pull/9130 merged, this issue can be closed.
